core/src/main/java/io/aiven/klaw/service/BaseOverviewService.java

```diff
@@ -224,9 +223,19 @@ protected List<AclOverviewInfo> applyFiltersAclsForSOT(
 
     List<AclOverviewInfo> aclList = new ArrayList<>();
     AclOverviewInfo mp;
+    List<AclRequests> disableShowDeleteAcl;
 
     for (Acl aclSotItem : aclsFromSOT) {
+      disableShowDeleteAcl =
+          aclRequestsRepo.findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
+              tenantId,
+              aclSotItem.getEnvironment(),
+              RequestStatus.DELETED.value,
+              aclSotItem.getTopicname());
```

Comment:
why should call to database happened before `if` and the result of this call is used only inside `if`?
I would suggest to make such call only if we already inside the `if`

Or another idea: can we have only one call to DB instead of making a call per loop iteration?

Reply:
Yes, I agree having only DB call would increase the speed drastically thanks for you suggestion. I will try to implement.

Reply:
The environment and topicname are common through the entire array so you can change this pretty easily.
One thought based on @snuyanzin comment (feel free to correct me here) is we could build the query only to return the ACLID for each matching row.
 List<AclID> findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
      int tenantId, String env, String requestStatus, String topicname);
      
      and then here something like (obvs check if the lists are empty etc before getting the envs)
      
          Set<Integer> set = aclRequestsRepo.findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
        tenantId,
        aclsFromSOT.get(0).getEnvironment(),
        RequestStatus.DELETED.value,
        aclsFromSOT.get(0).getTopicname()).stream().map(AclID::getReq_no).collect(Collectors.toSet());

   OR and Sergey knows Im not a huge fan you could use a custom @Query to select just the req_no column and return a set of Integers.

Reply:
Let me know if any of that is not clear.

Reply:
> 
aclsFromSOT.get(0).getTopicname()).stream().map(AclID::getReq_no).collect(Collectors.toSet());


> The environment and topicname are common through the entire array so you can change this pretty easily. One thought based on @snuyanzin comment (feel free to correct me here) is we could build the query only to return the ACLID for each matching row. List findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname( int tenantId, String env, String requestStatus, String topicname);
> 
> ```
>   and then here something like (obvs check if the lists are empty etc before getting the envs)
>   
>       Set<Integer> set = aclRequestsRepo.findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
>     tenantId,
>     aclsFromSOT.get(0).getEnvironment(),
>     RequestStatus.DELETED.value,
>     aclsFromSOT.get(0).getTopicname()).stream().map(AclID::getReq_no).collect(Collectors.toSet());
> ```
> 
> OR and Sergey knows Im not a huge fan you could use a custom @query to select just the req_no column and return a set of Integers.

aclsFromSOT.get(0).getTopicname()).stream().map(AclID::getReq_no).collect(Collectors.toSet());
this should gives an error cause AclID does not any getters did you mean
aclsFromSOT.get(0).getTopicname()).stream().map(AclRequests::getReq_no).collect(Collectors.toSet()); ?

Reply:
ACLID is using lombok annotations to create getters and setters, you might need a plugin to get your IDE to work with them, but intellij community edition I know at least does work with them. So that should not be a problem.

However I'm just rethinking this as I am looking at it with fresh eyes.

You will need to get the ACLRequest back because you need to get the otherParam entry not the AclID.

As the AclID is always unique and the otherParam is what we actually need. Sorry about that, i missed in it when I was looking at it the last day.

So get back the entire request where it is a delete type and the status is still create in the same env etc. then

Set<String> openDeleteRequests = aclsFromSOT.stream().map(AclRequest::otherParam).collect(Collectors.toSet());

openDeleteRequest.contains(Integer.toString(requestNum));